template/src/screens/HomeComponent/HomeScreen.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import ScreenContainer from '../../components/ScreenContainer'
+import {HomeStackParamList} from '../../navigation/types'
+import {StackNavigationProp} from '@react-navigation/stack'
+import {RouteProp} from '@react-navigation/native'
+
+interface IProps {
+  navigation: StackNavigationProp<HomeStackParamList, 'HomeScreen'>
+  route: RouteProp<HomeStackParamList, 'HomeScreen'>
```

Comment:
use `RouteKey` constants

Reply:
RouteName of StackNavigationProp only accept key of HomeStackParamList

Reply:
Then we can use
```
interface IProps {
  navigation: StackNavigationProp< HomeStackParamList, keyof HomeStackParamList >
  route: RouteProp< HomeStackParamList, keyof HomeStackParamList >
}
```

Reply:
outdated